charts/argo-cd/values.yaml

```diff
@@ -847,6 +847,58 @@ dex:
   # - secretRef:
   #     name: secret-name
 
+  # TLS certificate configuration via cert-manager
+  ## Ref: https://argo-cd.readthedocs.io/en/stable/operator-manual/tls/#configuring-tls-to-argocd-dex-server
+  certificate:
+    # -- Deploy a Certificate resource (requires cert-manager)
+    ## Note: Dex server must be restarted after certificate change, consider solutions like [Roloader] to watch for secret changes
```

Comment:
This is quite important and I would recommend to have [Reloader](https://github.com/stakater/Reloader) to avoid issues with cert rotation.

Reply:
Maybe we could file an Issue in the upstream project. The bundled dex component is controlled by a helper utility `/shared/argocd-dex rundex`

Reply:
For Dex I wanted to talk to Michael and propose / ask few questions. I believe the helper utility should not be used as there are another cleaner ways how to manage Dex and allow users to have HA setup, persistent sessions, etc. which is not currently possible with a way it's implemented.